### ISSUE TITLE
Break on Create Block Grid with XYZ

### ISSUE THREAD

**Issue by PitchPoler** [CSIRO-Precision-Agriculture/PAT_QGIS_Plugin#55]
Please forgive my inexperience with PAT, git and everything here.  I just hope to contribute a tiny fix I made in what I was trying to do.

I just started using PAT (version 1.0.3 on QGIS ver 3.16.2-Hannover on Windows10).  I struggled with an error in Create Block Grid:

![image](https://user-images.githubusercontent.com/77353148/104360365-8a99c580-54d6-11eb-8014-0eb3aba424a0.png)

I figured out that the problem was that I had an XYZ layer with Google Satellite baselayer in this project.  That does not have a 'path' key.  When no 'path' key in the variable named 'result' then this error is thrown by the get_layer_source(layer) function.  So I edited line 251 of util\qgis_common.py:
```
 if len(result) == 0 or (not 'path' in result) or result['path'] == '':
        return layer.source()
    else:
        return result['path']
```
That is the first thing I ever wrote in python, so you can make it better I'm sure.  But with that change I finally got 'Block grid completed successfully!'

Thank you for this tool.

**Comment by cratcliff** [comment]
@PitchPoler Thanks for the error report and suggestion

I have released a new version PAT v1.0.3 which contains a similar fix, whereby I look for the _path_ key in the results dictionary.
```   
 if not 'path' in result.keys():
        return ''
```

**Comment by PitchPoler** [comment]
Yes, your code is better, but I don't see it in the latest version here.  I am a git novice, so maybe I'm not looking at the right thing but here is the code block I am talking about from version 1.0.3 util\qgis_common.py:
![image](https://user-images.githubusercontent.com/77353148/104471631-b5405880-5580-11eb-8eb7-f187eae810b3.png)

Thank you again
